clients/vault/src/system.rs

```diff
@@ -57,22 +74,78 @@ impl VaultService {
 		Self { spacewalk_parachain: spacewalk_parachain.clone(), config, shutdown }
 	}
 
+	/// Returns SCPMessageHandler which contains the thread to connect/listen to the Stellar
+	/// Node. See the oracle.rs example
+	async fn create_handler(&self) -> Result<ScpMessageHandler, Error> {
+		prepare_directories().map_err(|e| {
+			tracing::error!("Failed to create the SCPMessageHandler: {:?}", e);
+			Error::StellarSdkError
+		})?;
+
+		let is_public_net =
+			self.spacewalk_parachain.is_public_network().await.map_err(Error::from)?;
+
+		let tier1_node_ip =
+			if is_public_net { TIER_1_VALIDATOR_IP_PUBLIC } else { TIER_1_VALIDATOR_IP_TESTNET };
+
+		let network: &Network = if is_public_net { &PUBLIC_NETWORK } else { &TEST_NETWORK };
+
+		tracing::info!(
+			"Connecting to {:?} through {:?}",
+			std::str::from_utf8(network.get_passphrase().as_slice()).unwrap(),
+			tier1_node_ip
+		);
+
+		let secret = SecretKey::from_encoding(&self.config.stellar_vault_secret_key).unwrap();
+		let public_key_binary = hex::encode(secret.get_public().as_binary());
+		tracing::info!("public key binary {:?}", public_key_binary);
+
+		let pk =
+			PublicKey::from_encoding("GA5ZSEJYB37JRC5AVCIA5MOP4RHTM335X2KGX3IHOJAPP5RE34K4KZVN")
+				.unwrap();
+		tracing::info!("mainnet {:?}", pk.as_binary());
+		let pk =
+			PublicKey::from_encoding("GAKNDFRRWA3RPWNLTI3G4EBSD3RGNZZOY5WKWYMQ6CQTG3KIEKPYWAYC")
+				.unwrap();
+		tracing::info!("testnet {:?}", pk.as_binary());
```

Comment:
You can remove this again, I just used it to find out the binaries for the stellar issuers of some USDC on stellar haha. No need to keep it. The values are used in the testchain configuration of the (not yet merged) vault communication branch.